src/main/java/io/github/jsonSnapshot/matchrule/SnapshotMatchRule.java

```diff
@@ -0,0 +1,7 @@
+package io.github.jsonSnapshot.matchrule;
+
+import io.github.jsonSnapshot.SnapshotMatchException;
+
+public interface SnapshotMatchRule {
+  void match(String rawSnapshot, String currentObject) throws SnapshotMatchException;
```

Comment:
This interface is a nice addition.

A few thoughts:
1. `SnapshotMatchingPolicy` or `SnapshotMatchingStrategy` might be a better name, as it might encompass multiple rules.
2. Consider moving this interface to the top-level package and leaving all its implementation details (different matching policies) in `matchingpolicy` or similar package (named to follow the interface). See [argument for this packaging strategy](https://javadevguy.wordpress.com/2017/12/18/happy-packaging/)
3. Consider passing `SnapshotDataItem`s into this method. This way it's more flexible, as in the future `SnapshotDataItem` might hold extra information that would be relevant for matching.

Reply:
doing this